### PR TITLE
Add skeletons of `mul`, `rowwise_add`, `sigmoid`, `softmax` ops

### PR DESCRIPTION
**Pull request by reyoung**
* Implement InferShape and register them, give a stub Kernel method
  by LOG(INFO)

### CODE REVIEW

File: paddle/framework/ddim.cc [hunk 1]
```diff
@@ -278,5 +278,11 @@ std::ostream& operator<<(std::ostream& os, const DDim& ddim) {
   return os;
 }
 
+ssize_t DDim::size() const { return product(*this); }
+
+DDim::DDim(std::initializer_list<int> init_list) {
```
Comment (reyoung) [inline paddle/framework/ddim.cc]: This constructor could be significantly easier for Op developer to invoke `Tensor::set_dims`.

Just by
```cpp
tensor.set_dims({100, 784})
```

File: paddle/framework/ddim.cc [hunk 2]
```diff
@@ -278,5 +278,11 @@ std::ostream& operator<<(std::ostream& os, const DDim& ddim) {
   return os;
 }
 
+ssize_t DDim::size() const { return product(*this); }
```
Comment (reyoung) [inline paddle/framework/ddim.cc]: Also, `size()` should be a member function for DDim. Invoke `product` is pretty strange to get the size.

File: paddle/operators/rowwise_add_op.cc [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+class RowWiseAddOp : public framework::OperatorWithKernel {
+protected:
+  void InferShape(
+      const std::vector<const framework::Tensor *> &inputs,
+      const std::vector<framework::Tensor *> &outputs) const override {
+    PADDLE_ENFORCE(inputs.size() == 2UL, "Two inputs is needed by rowwise add");
+    auto dim0 = inputs[0]->dims();
+    auto dim1 = inputs[1]->dims();
+
+    PADDLE_ENFORCE(dim0.size() == 2, "Input 0 must be matrix");
```
Comment (QiJune) [inline paddle/operators/rowwise_add_op.cc]: Here, a matrix is a tensor with rank=2. It should be
```
PADDLE_ENFORCE(paddle::framework::arity(dim0)==2, "input 0 must be matrix");
```

File: paddle/operators/mul_op.cc [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+class MulOp : public framework::OperatorWithKernel {
+protected:
+  void InferShape(
+      const std::vector<const framework::Tensor *> &inputs,
+      const std::vector<framework::Tensor *> &outputs) const override {
+    PADDLE_ENFORCE(inputs.size() == 2, "The mul op must take two inputs");
+    auto dim0 = inputs[0]->dims();
+    auto dim1 = inputs[1]->dims();
+    PADDLE_ENFORCE(dim0.size() == 2 && dim1.size() == 2,
```
Comment (QiJune) [inline paddle/operators/mul_op.cc]: It's not the size of dim, but rank of dim.

File: paddle/operators/rowwise_add_op.cc (hunk 3 above)
Comment (reyoung) [inline paddle/operators/rowwise_add_op.cc]: Done. Change size() to `arity`

File: paddle/operators/mul_op.cc (hunk 4 above)
Comment (reyoung) [inline paddle/operators/mul_op.cc]: Done.

File: paddle/operators/CMakeLists.txt [hunk 5]
```diff
@@ -42,3 +42,8 @@ endfunction()
 
 op_library(add_op SRCS add_op.cc add_op.cu)
 cc_test(add_op_test SRCS add_op_test.cc DEPS add_op)
+
+op_library(mul_op SRCS mul_op.cc mul_op.cu)
+op_library(rowwise_add_op SRCS rowwise_add_op.cu rowwise_add_op.cc)
+op_library(sigmoid_op SRCS sigmoid_op.cu sigmoid_op.cc)
+op_library(softmax_op SRCS softmax_op.cc softmax_op.cu)
```
Comment (QiJune) [inline paddle/operators/CMakeLists.txt]: It's better to maintain a consistent sequence. .cc first and .cu follows.

File: paddle/operators/mul_op.cc [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License. */
+
+#include <paddle/framework/op_registry.h>
+#include <paddle/framework/tensor.h>
+#include <paddle/operators/mul_op.h>
```
Comment (QiJune) [inline paddle/operators/mul_op.cc]: Please follow the code style, using 
```
#include "paddle/framework/op_registry.h"
#include "paddle/framework/tensor.h"
#include "paddle/operators/mul_op.h"
```

File: paddle/operators/CMakeLists.txt (hunk 5 above)
Comment (reyoung) [inline paddle/operators/CMakeLists.txt]: Just a unit-test here to test op_library is not caring about source sequence.

**Pull request merged**